src/Raven.Server/Rachis/RachisConsensus.cs

```diff
@@ -346,16 +346,18 @@ private void SetNewStateInTx(TransactionOperationContext context, State state, I
 
             context.Transaction.InnerTransaction.LowLevelTransaction.OnCommit += tx =>
             {
-                Task.Run(() =>
+                var tcs = Interlocked.Exchange(ref _stateChanged, new TaskCompletionSource<object>());
+                
+                Task.Factory.StartNew(t =>
                 {
-                    Interlocked.Exchange(ref _stateChanged, new TaskCompletionSource<object>()).TrySetResult(null);
-                    foreach (var t in toDispose)
+                    ((TaskCompletionSource<object>)t).TrySetResult(null);
+
+                    foreach (var d in toDispose)
                     {
-                        t.Dispose();
+                        d.Dispose();
                     }
-                });
+                }, tcs, CancellationToken.None, TaskCreationOptions.PreferFairness | TaskCreationOptions.RunContinuationsAsynchronously | TaskCreationOptions.LongRunning, TaskScheduler.Default);
```

Comment:
`LongRunning` is forcing the TP to create a new thread, meaning that the problem is contention.
We need to fix this somehow. Maybe have a dedicated thread just for those kind of things?